src/erlcloud_s3.erl

```diff
@@ -1156,3 +1165,35 @@ aws_region_from_host(Host) ->
         _ ->
             "us-east-1"
     end.
+
+%%
+%% http://docs.aws.amazon.com/AmazonS3/latest/dev/VirtualHosting.html
+%% http://docs.aws.amazon.com/AmazonS3/latest/dev/Redirects.html
+%% Note: redirect responce is handled only once.
+s3_follow_redirect({error, {http_error, _StatusCode, _StatusLine, ErrBody, ErrHeaders}} = Response, 
+    Config, Method, Bucket, Path, Subresource, Params, POSTData, Headers) ->
```

Comment:
may be move `when StatusCode >= 301 andalso StatusCode < 400` as guards here or even have 
s3_follow_redirect( fun s3_request4_no_update/8, Config1, Method, Bucket, Path, Subresource, Params, POSTData, Headers);